torchbox/torchbox.com
# JS implementation of mode switcher

[Link to Ticket]()

### Description of Changes Made

Implement the site switcher in JavaScript (based on the RNIB implementation).

- Adds an un-styled site-switcher form
- Adds JavaScript which sets a cookie to remember the user's preference, and updates the colour mode
- Adds settings to store the allowed values for the mode
- Adds a server-side context processor to read the value from the cookie, and initially set the correct classes and data attribute based on the cookie preference, if it exists. Doing it this way avoids a FOUC
- The user's preferred setting for light or dark mode is not currently respected, as we want to use dark mode by default.

To come: a fallback method to set the cookie on the server-side if the user has JavaScript disabled.

Documentation updates will be done once the whole light mode feature is complete.

### How to Test

- Load any page in your local build
- Make a selection for dark or light mode from the un-styled form in the header
- Ensure the styles instantly update
- Ensure a 'mode' cookie has been set (check the 'Applications' section of dev tools to view cookies)
- Ensure your selection is remembered as you browse around the site

### Screenshots

<details>
  <summary>Expand to see more</summary>

![Screenshot 2024-07-02 at 15 17 38](https://github.com/torchbox/torchbox.com/assets/771869/fb925d69-c334-4363-9ad5-98eb928e4c6f)

</details>

### MR Checklist


#### Unit tests


#### Documentation


#### Browser testing

  - Latest version of Chrome on mac
  - Latest version of Firefox on mac
  - Latest version of Safari on mac
  - Safari on last two versions of iOS
  - Chrome on last two versions of Android

#### Data protection


#### Accessibility


#### Sustainability


#### Pattern library
